API and Gateway v8

msciotti opened the pull request:
TODO:

- [x] Document deprecated routes
- [x] ~~Complete error code documentation for API~~ lol jk there's a million

DO LATER:

- [ ] Integrations optional values
- [ ] Table spacing
- [ ] Pick int or integer and just use it everwhere

msciotti committed:
v8 changes

night commented on docs/resources/Guild.md:
```diff
@@ -15,13 +15,12 @@ Guilds in Discord represent an isolated collection of users and channels, and ar
 | afk_timeout                   | integer                                                                             | afk timeout in seconds                                                                                                                    |
-| embed_enabled?                | boolean                                                                             | true if the server widget is enabled (deprecated, replaced with `widget_enabled`)                                                         |
-| embed_channel_id?             | ?snowflake                                                                          | the channel id that the widget will generate an invite to, or `null` if set to no invite (deprecated, replaced with `widget_channel_id`)  |
+| widget_enabled?                | boolean                                                                             | true if the server widget is enabled (deprecated, replaced with `widget_enabled`)                                                         |
```
these are now in this list twice

advaith1 replied:
also on lines 172-173

SinisterRectus commented on docs/topics/Rate_Limits.md:
````diff
@@ -75,85 +75,4 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
-}
-```
-
-## Invalid Request Limit
````
(Moving the previous comment to here.) Did you mean to remove this section?

msciotti replied:
Not this one, woops!

night commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,24 @@
+#### September 24, 2020
+
+We've introduced API and Gateway v8! Changes are noted throughout the documentation, but the major changes are:
+- API v8 is now default
```
v8 is not yet default. we'll need to have a deprecation period before it hits default

night commented on docs/topics/Gateway.md:
```diff
@@ -9,7 +9,8 @@ Important note: Not all event fields are documented, in particular, fields prefi
-| ------- | ------------ |
+|---------|--------------|
+| 8       | Available    |
 | 6       | Available    |
```
we should document v6 as deprecated

night commented on docs/topics/Opcodes_and_Status_Codes.md:
```diff
@@ -1,100 +1,114 @@
+| BASE_TYPE_CHOICES    | "Value must be one of {choices}."                          |
+| BASE_TYPE_MIN_LENGTH | "Must be {min_length} or more in length."                  |
+| BASE_TYPE_MAX_LENGTH | "Must be {max_length} or fewer in length."                 |
+| BASE_TYPE_BAD_LENGTH | "Must be between {min_length} and {max_length} in length." |
```
this is not a complete collection of error codes, are we posting these just as examples?

msciotti replied:
Oh I went into the constants file and though that was the lot of them? I'll look more

msciotti replied:
JK there's a million not doing that

night commented on docs/topics/Permissions.md:
```diff
@@ -168,8 +170,7 @@ Roles represent a set of permissions attached to a group of users. Roles have un
 | position        | integer   | position of this role                            |
-| permissions     | integer   | legacy permission bit set                        |
-| permissions_new | string    | permission bit set                               |
+| permissions     | string    | legacy permission bit set                        |
```
rm legacy

night commented on docs/topics/Rate_Limits.md:
```diff
@@ -40,7 +40,7 @@ In the case that a rate limit is exceeded, the API will return a HTTP 429 respon
 | ----------- | ------- | --------------------------------------------------------------------- |
 | message     | string  | A message saying you are being rate limited.                          |
-| retry_after | integer | The number of milliseconds to wait before submitting another request. |
+| retry_after | integer | The number of seconds to wait before submitting another request. |
```
this is now a integer or float

night commented on docs/topics/Rate_Limits.md:
```diff
@@ -75,85 +75,4 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
   "retry_after": 6457,
```
we need to update these examples.

retry-after is now in seconds, not ms

msciotti replied:
Is `64.57` valid since it can be a float

night replied:
`64.57` and `65` would be the header

advaith1 commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,24 @@
+- Form body errors have been improved to include more helpful messaging on validation. [See more here](#DOCS_TOPICS_OPCODES_AND_STATUS_CODES/API)
+- Rate limit headers are in seconds instead of milliseconds. As such, the `X-RateLimit-Precision header` is no longer respected. If you receive any millisecond headers, please report it as a bug!
+- Bots no longer receive [Channel Create Gateway Event](#DOCS_GATEWAY/channel-create) for DMs
+- `delete-message-days` is no longer available. Use `delete_message_days`. The previous value may never have been documented
```
was documented and updated in b9edace323c9df64c79f104d85984690ae4e2977

msciotti committed:
Deprecated routes

msciotti committed:
PR fixes
- Fix widget_ values
- v8 not default
- v6 gateway deprecated

msciotti committed:
More PR fixes
- Document widget.json
- Fix widget/embed removed in favor of things
- Remove double values
- Rate limit example update

msciotti committed:
Retry after examples

msciotti committed:
Unrelated, spelling is hard

msciotti committed:
No legacy permissions

msciotti committed:
Accidental section delete

msciotti committed:
Changelog update

advaith1 commented on docs/resources/Guild.md:
```diff
@@ -785,17 +777,46 @@ Sync an integration. Requires the `MANAGE_GUILD` permission. Returns a 204 empty
 Modify a [guild widget](#DOCS_RESOURCES_GUILD/guild-widget-object) object for the guild. All attributes may be passed in with JSON and modified. Requires the `MANAGE_GUILD` permission. Returns the updated [guild widget](#DOCS_RESOURCES_GUILD/guild-widget-object) object.
 
-## Modify Guild Embed % PATCH /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-object}/embed
+## Get Guild Widget JSON % GET /guilds/{guild.id#DOCS_RESOURCES_GUILD/guild-bject}/widget.json
```
1. bject in url should be object
2. I would name this just "Get Guild Widget" and rename the other routes to "... Widget Settings" (`get_widget_settings` internally)

tpcstld committed:
Some changelog cleanup.

msciotti committed:
Deprecated and default lul

msciotti committed:
Fix widget routes

advaith1 commented on docs/resources/Guild.md:
````diff
@@ -781,21 +773,59 @@ Delete the attached [integration](#DOCS_RESOURCES_GUILD/integration-object) obje
+    "enabled": true,
+    "channel_id": "639513359995470208"
+}
+```
````
duplicate of https://discord.com/developers/docs/resources/guild#guild-widget-object

advaith1 commented on docs/Reference.md:
```diff
@@ -18,12 +18,67 @@ Discord exposes different versions of our API. You can specify which version to
-| 6       | Available    | ✓       |
+|---------|--------------|---------|
+| 8       | Available    |         |
+| 6       | Deprecated   | ✓       |
```
maybe document v7 here as deprecated?

msciotti replied:
v7 was technically never released. @night do we want to reference it or no

msciotti committed:
That is way too many error codes

msciotti committed:
Dupe return example

msciotti committed:
Just for you, night

tpcstld committed:
typos.

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -388,30 +388,28 @@ See [permissions](#DOCS_TOPICS_PERMISSIONS/permissions) for more information abo
+| id    | snowflake | role or user id               |
+| type  | int       | either 0 (role) or 1 (member) |
+| allow | string    | permission bit set            |
+| deny  | string    | permission bit set            |
```
If permissions can't be sent as ints now, then these tables shouldn't be duplicated anymore, as they are the same now (I only separated these when _new fields were added)

advaith1 replied:
tpcstld says permissions can still be sent as ints or strings, but this PR changes it (in multiple places) to only say they can be sent as strings; Is sending it as ints considered deprecated or was this mistakenly changed?

tpcstld replied:
@msciotti we might want to talk about this :'D

* For the time being, the API is definitely going to support both JSON int and strings, since we just cast it to a number internally.
* Nevertheless, our official clients will only ever send strings.
* It is entirely up to you whether or not we want to publicly deprecate ints here.

My 2 cents is maybe keep ints here and deprecate in the future if we have to? /shrug Or say that we prefer strings here?

Sei4or commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,42 @@
+- `/servers/<guild_id>/widget.json`
+- `/guilds/<guild_id>/embed.json`
+
+Removed in favor if `/guilds/<guild_id>/widget.png`:
```
Typo. Should be `of` instead of `if`.

tpcstld committed:
More typos.

cloudrac3r commented on docs/topics/Rate_Limits.md:
```diff
@@ -50,15 +50,15 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
 {
   "message": "You are being rate limited.",
-  "retry_after": 6457,
+  "retry_after": 64.57,
```
These aren't equivalent. Did you mean 6.457?

night replied:
it's an example value. it doesn't need to be the exact same number

wasdennnoch replied:
If the rate limits have millisecond-precision I suggest also showcasing said precision in the example. Sure, this is a valid value, but just to be sure everyone gets that there can be up to three decimals I'd use `6.457` or `64.570`.

vladfrangu commented on docs/resources/Channel.md:
```diff
@@ -857,24 +855,20 @@ Any message IDs given that do not exist or are invalid will count towards the mi
+|-------|--------|-------------------------------------------------|
+| allow | string | the bitwise value of all allowed permissions    |
+| deny  | string | the bitwise value of all disallowed permissions |
+| type  | string | "member" for a user or "role" for a role        |
```
Why is this still a string, when the API returns numbers? Is that intended, or is that just an oversight? 👀

advaith1 replied:
docs oversight, should say number

msciotti replied:
Fixed in latest

wasdennnoch commented on docs/topics/Rate_Limits.md:
```diff
@@ -37,11 +37,11 @@ In the case that a rate limit is exceeded, the API will return a HTTP 429 respon
+| Field       | Type             | Description                                                      |
+|-------------|------------------|------------------------------------------------------------------|
+| message     | string           | A message saying you are being rate limited.                     |
+| retry_after | integer or float | The number of seconds to wait before submitting another request. |
```
While some JSON parsers themselves might be able to automatically detect the numeric type, it is still incredibly helpful for library/bot devs to know which fields are integers and which are floats without having to manually inspect every possible response body. This is especially important for more strictly typed languages. The JSON parser is only a small part of the entire library/bot.

SinisterRectus replied:
Yeah this is weird. If it can have an important fractional part, it should be `float` or `number`.

tpcstld replied:
```suggestion
| retry_after | float | The number of seconds to wait before submitting another request. |
```

In most (all?) cases it's a number with decimals.

wasdennnoch commented on docs/topics/Rate_Limits.md:
```diff
@@ -50,15 +50,15 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
 < X-RateLimit-Limit: 10
 < X-RateLimit-Remaining: 0
 < X-RateLimit-Reset: 1470173023
 < X-RateLimit-Reset-After: 7
```
I think you forgot to update this header to `64.57`. Possibly also the one above that (to something like `X-RateLimit-Reset: 1470173023.420`)

wasdennnoch replied:
I was referring to `X-RateLimit-Reset-After`; `Retry-After` is indeed supposed to be in rounded settings.

SinisterRectus commented on docs/Reference.md:
```diff
@@ -17,12 +17,68 @@ Discord exposes different versions of our API. You can specify which version to
+        "activities": {
+            "0": {
+                "platform": {
+                    "_errors": [
```
The underscore distinguishes the error field from the actual field names.

SinisterRectus replied:
Probably should be, with this concept re-visited: https://github.com/discord/discord-api-docs/pull/967/files#diff-b307d936b736f302bc7cc153f0d8a0a3R7

msciotti replied:
@tpcstld should we expect libraries to ignore/pass through the `_error` field? We do say that `_` prefixed fields are internal only, so if that's the case, I can un-document this.

tpcstld replied:
This is a great question. @night might be better equipped as he created this error API, but I believe this is one of the cases where we have an `_` prefix but the field is public. Woe is API design.

The reason why the `_` exists is to avoid a possible collision with a "legitimate" field named "error", due to the nested nature of the object. My opinion is that, to support bot users who may wish to return custom errors, we can have this be an exception to the rule.

advaith1 replied:
anyway, that note about underscore fields is for gateway so it doesn't apply to this

MinnDevelopment commented on docs/topics/Rate_Limits.md:
```diff
@@ -50,15 +50,15 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
 < X-RateLimit-Limit: 10
 < X-RateLimit-Remaining: 0
 < X-RateLimit-Reset: 1470173023
 < X-RateLimit-Reset-After: 7
```
This example is kinda inconsistent, i suggest this change:
```suggestion
< Retry-After: 7
< X-RateLimit-Limit: 10
< X-RateLimit-Remaining: 0
< X-RateLimit-Reset: 1470173023.123
< X-RateLimit-Reset-After: 6.457
```

advaith1 commented on docs/resources/Guild.md:
```diff
@@ -15,13 +15,12 @@ Guilds in Discord represent an isolated collection of users and channels, and ar
 | owner_id                      | snowflake                                                                           | id of owner                                                                                                                               |
-| permissions?                  | integer                                                                             | legacy total permissions for [the user](#DOCS_RESOURCES_USER/get-current-user-guilds) in the guild (excludes overrides)                   |
-| permissions_new?              | string                                                                              | total permissions for [the user](#DOCS_RESOURCES_USER/get-current-user-guilds) in the guild (excludes overrides)                          |
+| permissions?                  | string                                                                             | total permissions for [the user](#DOCS_RESOURCES_USER/get-current-user-guilds) in the guild (excludes overrides)                   |
```
the description says `the user` and links to the only endpoint where it is returned

tpcstld replied:
IMO we might refactor this part of the doc in the future, but for now I think we should leave it as-is since its description is aligned with other parts of this table (e.g. `owner`)

I also got bamboozled by this btw XD

advaith1 commented on docs/topics/Gateway.md:
```diff
@@ -679,7 +680,7 @@ The inner `d` key is a boolean that indicates whether the session may be resumab
 #### Channel Create
 
-Sent when a new channel is created, relevant to the current user. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object.
+Sent when a new channel is created, relevant to the current user. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object. Does not fire for DMs.
```
maybe
```suggestion
Sent when a new guild channel is created, relevant to the current user. The inner payload is a [channel](#DOCS_RESOURCES_CHANNEL/channel-object) object.
```

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -29,15 +29,15 @@ Represents a guild or DM channel within Discord.
+| GUILD_STORE    | 6  | a channel in which game developers can [sell their game on Discord](https://discord.com/developers/docs/game-and-server-management/special-channels) |
 
 ###### Example Guild Text Channel
 
```
I believe that it's only displayed as a single object because it's internally represented as a single object, there are examples for each channel type though

ByteAlex commented on docs/topics/Rate_Limits.md:
```diff
@@ -68,92 +68,23 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
 
-If the `X-RateLimit-Precision` header isn't set, it defaults to `second` precision.
+- **401** responses are avoided by providing a valid token in the authorization header when required and by stopping further requests after a token becomes invalid	
+- **403** responses are avoided by inspecting role or channel permissions and by not making requests that are restricted by such permissions	
```
Is there any option to avoid these for DM channels?
My whole logging is spammed with DM failures, though everything else is catched.

SinisterRectus replied:
Not preemptively, no. If you have a large application that relies on DMs, you will probably want to track which channels give you a 403 and not make further requests to those. You can also consider self-imposing a global ratelimit on DMs.

ByteAlex replied:
I know that it wasn't possible yet, but this feature is really needed, - so take it as a hint for a missing feature.

advaith1 replied:
a failure to create a DM is a 400, not a 403, so it doesn't contribute toward this limit (also its not related to v8)

SinisterRectus replied:
> I know that it wasn't possible yet, but this feature is really needed, - so take it as a hint for a missing feature.

You should open a separate issue for that.

ByteAlex replied:
> Are you bulk DMing people for this to happen or is it just a really big bot that responds to help commands in DMs or something, if you don't mind me asking?

My bot is in about 550k servers. Trust me, I'm familar with the discord api and the (API) ToS.
Some features which send DM messages:
- Reminders
- Reaction-Roles (DM confirmation after role (un)assignment)
- Lottery Results
- Help Messages
- Vote Reminders
- Vote Messages (after a successful vote you'll get a DM confirmation, about 20k/day)

I think this list is already sufficient to show that my DM's are "fine".

msciotti commented on docs/topics/Opcodes_and_Status_Codes.md:
```diff
@@ -6,95 +6,95 @@ All gateway events in Discord are tagged with an opcode that denotes the payload
+| 200 (OK)                  | The request completed successfully.                                                  |
+| 201 (CREATED)             | The entity was created successfully.                                                 |
+| 204 (NO CONTENT)          | The request completed successfully but returned no content.                          |
+| 304 (NOT MODIFIED)        | The entity was not modified (no action was taken).                                   |
```
I can look into this

msciotti committed:
Link to gateway version table

Co-authored-by: Nekokatt <nekoka.tt@outlook.com>

msciotti committed:
Non-dm --> guild

Co-authored-by: Nekokatt <nekoka.tt@outlook.com>

msciotti committed:
Clarify channel create

Co-authored-by: Advaith <advaithj1@gmail.com>

msciotti committed:
Fix rate limit consistency example

Co-authored-by: Florian Spieß <business@minnced.club>

msciotti commented on README.md:
```diff
@@ -22,4 +22,4 @@ Alert boxes are achieved by using a block quote that has one of 'warn', 'danger'
 The Unofficial Discord API server is a common hangout for library and bot developers alike. It's a great starting point for those looking to dive in and learn bot-creation with the Discord API.
 
-[![](https://discord.com/api/guilds/81384788765712384/embed.png?style=banner1)](https://discord.gg/discord-api)
+[![](https://discord.com/api/guilds/81384788765712384/widget.png?style=banner1)](https://discord.gg/discord-api)
```
Thank you, great callout. Realistically all the images in the docs should have this too.

msciotti committed:
Add alt text for DAPI widget

Co-authored-by: Nekokatt <nekoka.tt@outlook.com>

msciotti committed:
Cosistency of versioning and overwrite type

advaith1 commented on docs/topics/Gateway.md:
```diff
@@ -96,8 +97,8 @@ def on_websocket_message(msg):
 
 | Field     | Type    | Description                                   | Accepted Values                                                            |
-| --------- | ------- | --------------------------------------------- | -------------------------------------------------------------------------- |
-| v         | integer | Gateway Version to use                        | 6 (see [Gateway versions](#DOCS_TOPICS_GATEWAY/gateways-gateway-versions)) |
+|-----------|---------|-----------------------------------------------|----------------------------------------------------------------------------|
+| v         | integer | Gateway Version to use                        | see [Gateway versions](#DOCS_TOPICS_GATEWAY/gateways-gateway-versions) |
 | encoding  | string  | The encoding of received gateway packets      | 'json' or 'etf'                                                            |
 | compress? | string  | The (optional) compression of gateway packets | 'zlib-stream'                                                              |
```
fix table spacing

msciotti replied:
My autoformatter is really messing me up. I'm just gonna turn it off for now and do a separate pass in the future for spacing.

msciotti commented on docs/topics/Opcodes_and_Status_Codes.md:
```diff
@@ -6,95 +6,95 @@ All gateway events in Discord are tagged with an opcode that denotes the payload
+| 7    | Resume              | client            | Resume a connection.                                     |
+| 8    | Hello               | server            | Time to wait between sending heartbeats in milliseconds. |
+| 9    | Resumed             | server            | Acknowledge a successful session resume.                 |
+| 13   | Client Disconnect   | server            | A client has disconnected from the voice channel         |
```
It's been a while since I've dug into the Voice API tbh. Last I remember was something like it still being touched very heavily and not wanting to document in case we make a lot of breaking changes. But I can bring this back up next week as a separate note.

night replied:
we already now send 443 in the endpoints as of a couple weeks ago

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -388,30 +388,28 @@ See [permissions](#DOCS_TOPICS_PERMISSIONS/permissions) for more information abo
+| Field | Type      | Description                   |
+|-------|-----------|-------------------------------|
+| id    | snowflake | role or user id               |
+| type  | int       | either 0 (role) or 1 (member) |
```
3 times in this file it says `int` but the rest of the docs say `integer`

msciotti replied:
We are so inconsistent in this. Main docs use integer, sdk uses int :( I'll rectify this in some other commit so it's not literally every file touched.

advaith1 commented on docs/resources/Audit_Log.md:
```diff
@@ -86,31 +86,31 @@ Whenever an admin action is performed on the API, an entry is added to the respe
+| message_id         | snowflake | id of the message that was targeted                         | MESSAGE_PIN & MESSAGE_UNPIN                                                    |
+| count              | string    | number of entities that were targeted                       | MESSAGE_DELETE & MESSAGE_BULK_DELETE & MEMBER_DISCONNECT & MEMBER_MOVE         |
+| id                 | snowflake | id of the overwritten entity                                | CHANNEL_OVERWRITE_CREATE & CHANNEL_OVERWRITE_UPDATE & CHANNEL_OVERWRITE_DELETE |
+| type               | int       | type of overwritten entity - 0 for "role" or 1 for "member" | CHANNEL_OVERWRITE_CREATE & CHANNEL_OVERWRITE_UPDATE & CHANNEL_OVERWRITE_DELETE |
```
this is actually an integer sent as a string (`"0"` or `"1"`); #2088

tpcstld replied:
Haha true.

This is a side-effect of an implementation detail that makes everything here string-serialized.

```suggestion
| type               | string       | type of overwritten entity - `"0"` for "role" or `"1"` for "member" | CHANNEL_OVERWRITE_CREATE & CHANNEL_OVERWRITE_UPDATE & CHANNEL_OVERWRITE_DELETE |
```

msciotti replied:
I've got this in a few places so I'll resolve them all

advaith1 replied:
its an int everywhere else, just a string here in audit log

tpcstld commented on docs/topics/Rate_Limits.md:
```diff
@@ -50,15 +50,15 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
 < HTTP/1.1 429 TOO MANY REQUESTS
 < Content-Type: application/json
-< Retry-After: 6457
+< Retry-After: 7
```
```suggestion
< Retry-After: 65
```

tpcstld commented on docs/topics/Rate_Limits.md:
```diff
@@ -50,15 +50,15 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
-< X-RateLimit-Reset: 1470173023
-< X-RateLimit-Reset-After: 7
+< X-RateLimit-Reset: 1470173023.123
+< X-RateLimit-Reset-After: 6.457
```
```suggestion
< X-RateLimit-Reset-After: 64.57
```

crowlKats commented on docs/topics/Gateway.md:
```diff
@@ -365,13 +367,13 @@ Events are payloads sent over the socket to a client that correspond to events i
 | [Invalid Session](#DOCS_TOPICS_GATEWAY/invalid-session)                             | failure response to [Identify](#DOCS_TOPICS_GATEWAY/identify) or [Resume](#DOCS_TOPICS_GATEWAY/resume) or invalid active session |
-| [Channel Create](#DOCS_TOPICS_GATEWAY/channel-create)                               | new channel created                                                                                                              |
+| [Channel Create](#DOCS_TOPICS_GATEWAY/channel-create)                               | new guild channel created                                                                                                       |
 | [Channel Update](#DOCS_TOPICS_GATEWAY/channel-update)                               | channel was updated                                                                                                              |
 | [Channel Delete](#DOCS_TOPICS_GATEWAY/channel-delete)                               | channel was deleted                                                                                                              |
```
Shouldn't it specify guild channel here as well?

advaith1 replied:
Does it not fire when DMs are closed?

crowlKats replied:
doesnt make much sense to have delete/close when we dont have channel create for dms either

SinisterRectus replied:
Get Channel still exists.

advaith1 commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,42 @@
+- All permissions have been converted to strings-serialized numbers. As such, `permissions_new`, `allow_new`, and `deny_new` have been removed
+- The `game` field has been removed. If you need a direct replacement, you can instead reference the first element of `activities`
+- Channel Permission Overwrite `type`s are now numbers (0 and 1) instead of strings ("role" and "member")
+- `embed_enabled` and `embed_channel_id` have been renamed `widget_enabled` and `widget_channel_id`
```
not really renamed, the `widget_` fields already existed
```suggestion
- `embed_enabled` and `embed_channel_id` have been removed. Use `widget_enabled` and `widget_channel_id` instead.
```

advaith1 commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,42 @@
+- Bots no longer receive [Channel Create Gateway Event](#DOCS_GATEWAY/channel-create) for DMs
+- `delete-message-days` is no longer available. Use `delete_message_days`.
+- Removed `roles`, `premium_since`, and `nick` from [Presence Update Gateway Event](#DOCS_GATEWAY/presence-update)
+- The following routes have been deprecated and removed for better naming conventions:
```
were already deprecated
```suggestion
- The following deprecated routes have been removed for better naming conventions:
```

advaith1 replied:
Also, `/invite/<invite_code>` was removed in favor of `/invites/<invite_code>`

advaith1 commented on docs/topics/Gateway.md:
```diff
@@ -819,15 +820,15 @@ You can use the `chunk_index` and `chunk_count` to calculate how many chunks are
+| not_found?  | array                                                                      | if passing an invalid id to `REQUEST_GUILD_MEMBERS`, it will be returned here               |
+| presences?  | array of [presence](#DOCS_TOPICS_GATEWAY/presence) objects                 | if passing true to `REQUEST_GUILD_MEMBERS`, presences of the returned members will be here  |
+| nonce?      | string                                                                     | the nonce used in the [Guild Members Request](#DOCS_TOPICS_GATEWAY/request-guild-members)   |
 
```
uh, looks like you commented on the Guild Members Chunk area, which is the response to Request Guild Members, which only needs the presence intent to return presences. If you were talking about Guild Create, that's [already documented](https://discord.com/developers/docs/topics/gateway#guild-create) in the right place.

msciotti committed:
Retry-After fix

Co-authored-by: Jerry Jiang <jerry.jiang@discordapp.com>

msciotti committed:
Reset-After fix

Co-authored-by: Jerry Jiang <jerry.jiang@discordapp.com>

msciotti committed:
renamed --> removed

Co-authored-by: Advaith <advaithj1@gmail.com>

msciotti committed:
Grammar nits

Co-authored-by: Advaith <advaithj1@gmail.com>

msciotti committed:
Deprecated old invite route, overwrites are strings

msciotti committed:
retry_after as a float for consistency

Co-authored-by: Jerry Jiang <jerry.jiang@discordapp.com>

msciotti committed:
Clarify user-relative fields in guild object

msciotti commented on docs/resources/Guild.md:
```diff
@@ -55,6 +55,8 @@ Guilds in Discord represent an isolated collection of users and channels, and ar
 
 ** \* These fields are only sent within the [GUILD_CREATE](#DOCS_TOPICS_GATEWAY/guild-create) event **
 
+** \*\* These fields are only sent when using the [GET Current User Guilds](#DOCS_RESOURCES_USER/get-current-user-guilds) endpoint and are relative to the requested user **
+
```
@nekokatt @tpcstld Is this a bit more clear for these two fields?

tpcstld replied:
I think this is good.

msciotti committed:
Fix overwrite types in changelog

tpcstld committed:
Channel permission overwrite.

queer commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,46 @@
+- Form body errors have been improved to include more helpful messaging on validation. [See more here](#DOCS_REFERENCE/error-messages)
+- The `Retry-After` header is now based in seconds instead of milliseconds (e.g. `123` means 123 seconds)
+- The `X-RateLimit-Precision` header is no longer respected. `X-RateLimit-Reset` and `X-RateLimit-Reset-After` are always returned at millisecond precision (e.g. `123.456` instead of `124`)
+- Bots no longer receive [Channel Create Gateway Event](#DOCS_GATEWAY/channel-create) for DMs
```
I'm curious, what's the reasoning behind this change? Was it just getting that expensive to send these to bots constantly?

advaith1 replied:
iirc gateway v6 sent a channel_create every time a dm was sent, even if in an existing dm

queer replied:
oooo right, I completely forgot about that >w<

advaith1 commented on docs/resources/Channel.md:
```diff
@@ -857,24 +855,20 @@ Any message IDs given that do not exist or are invalid will count towards the mi
+|-------|--------|-------------------------------------------------|
+| allow | string | the bitwise value of all allowed permissions    |
+| deny  | string | the bitwise value of all disallowed permissions |
+| type  | string | "0" for "role" or "1" for "member"                  |
```
its an int here, only string in audit log `options`
```suggestion
| type  | integer | 0 for a role or 1 for a member                  |
```

msciotti committed:
String only in audit logs

Co-authored-by: Advaith <advaithj1@gmail.com>

msciotti committed:
channel create no longer in dms

tpcstld commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,46 @@
+
+We've introduced API and Gateway v8! Changes are noted throughout the documentation, but the major changes are:
+
+- API v8 is now available. v6 is still the default for the time being.
```
```suggestion
- API and Gateway v8 is now available. v6 is still the default for the time being.
```

tpcstld commented on docs/Change_Log.md:
```diff
@@ -1,5 +1,46 @@
+- API v8 is now available. v6 is still the default for the time being.
+- All permissions have been converted to strings-serialized numbers. As such, `permissions_new`, `allow_new`, and `deny_new` have been removed
+- The `game` field has been removed. If you need a direct replacement, you can instead reference the first element of `activities`
+- Channel Permission Overwrite `type`s are now numbers (0 and 1) instead of strings ("role" and "member"). However due to a current technical constraint, they are string-serialized numbers in audit logs.
```
```suggestion
- Channel Permission Overwrite `type`s are now numbers (0 and 1) instead of strings ("role" and "member"). However due to a current technical constraint, they are string-serialized numbers in audit log `options`.
```

night commented on docs/topics/OAuth2.md:
```diff
@@ -299,15 +299,15 @@ When receiving the access code on redirect, there will be additional querystring
     "splash": null,
     "features": [],
     "afk_channel_id": null,
     "widget_enabled": false,
```
widget here twice

night commented on docs/topics/OAuth2.md:
```diff
@@ -299,15 +299,15 @@ When receiving the access code on redirect, there will be additional querystring
     "widget_channel_id": null,
```
widget here twice

msciotti committed:
t e c h n i c a l c o n s t r a i n t s

Co-authored-by: Jerry Jiang <jerry.jiang@discordapp.com>

night reviewed (approved):
🆗 minus duplicate widgets

msciotti committed:
PR nits
